exporter/loggingexporter/logging_exporter.go

```diff
@@ -174,6 +176,27 @@ func NewMetricsExporter(config configmodels.Exporter, logger *zap.Logger) (expor
 			// TODO: Add ability to record the received data
 			return 0, nil
 		},
-		exporterhelper.WithShutdown(logger.Sync),
+		exporterhelper.WithShutdown(loggerSync(logger)),
 	)
 }
+
+func loggerSync(logger *zap.Logger) func() error {
+	return func() error {
+		// Currently Sync() on stdout and stderr return errors on Linux and macOS,
+		// respectively:
+		//
+		// - sync /dev/stdout: invalid argument
+		// - sync /dev/stdout: inappropriate ioctl for device
+		//
+		// Since these are not actionable ignore them.
+		err := logger.Sync()
```

Comment:
I was tempted to simply remove `logger.Sync()` but its docs state that it should be called, so keeping it.